cmd/katapult/client_test.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/krystal/go-katapult"
+
+	"github.com/krystal/katapult-cli/config"
+)
+
+func TestNewClient_APIKey(t *testing.T) {
+	c, err := newClient(&config.Config{
+		APIKey: "test",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.(*katapult.Client).APIKey != "test" {
+		t.Fatal("API key is unset")
+	}
+}
+
+func TestNewClient_BaseURL(t *testing.T) {
+	// Test valid base URL.
+	c, err := newClient(&config.Config{
+		APIKey: "test",
+		APIURL: "https://example.com",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.(*katapult.Client).APIKey != "test" {
+		t.Fatal("API key is set to", c.(*katapult.Client).APIKey)
+	}
+	s := c.(*katapult.Client).BaseURL.String()
+	if s != "https://example.com" {
+		t.Fatal("invalid base URL:", s)
+	}
+
+	// Test invalid base URL.
+	_, err = newClient(&config.Config{
+		APIKey: "test",
+		APIURL: "this is a test",
+	})
+	if err == nil {
+		t.Fatal("expected error")
+	}
+}
```

Comment:
I think these two tests could benefit from being turned into a single `Test_newClient` tabular test that has test cases for setting each of the available attributes, and also one for setting all of the attributes.

Also I would recommend using [`testify/assert`](https://github.com/stretchr/testify#assert-package) for assertions to make things cleaner and easier. We tend to use it in pretty much all of our Go projects already :)